### PR TITLE
[MAINTENANCE] Remove legacy DataConnectors

### PR DESCRIPTION
**Pull request by joshua-stauffer**
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

**Review by cdkini — APPROVED** [review]

File: great_expectations/datasource/data_connector/util.py [hunk 1]
```diff
@@ -1,571 +0,0 @@
...
-                return None
-
-            if "name" not in sorter_config:
-                raise ValueError("Sorter config should have a name")  # noqa: TRY003
-
-            sorter_name: str = sorter_config["name"]
-            new_sorter: Sorter = _build_sorter_from_config(sorter_config=sorter_config)
-            sorter_dict[sorter_name] = new_sorter
-
-    return sorter_dict
```
Comment (joshua-stauffer) [inline great_expectations/datasource/data_connector/util.py:540]: if these utils were in use, the were migrated to the fluent dataconnector module where they are used

File: great_expectations/self_check/util.py [hunk 2]
```diff
@@ -913,27 +912,14 @@ def build_sa_validator_with_data(  # noqa: C901, PLR0912, PLR0913, PLR0915
...
-                )
-            )
-        )[0]
+        # maintain legacy behavior - standup a dummy LegacyBatchDefinition
+        batch_definition = LegacyBatchDefinition(
+            datasource_name="my_test_datasource",
+            data_connector_name="my_sql_data_connector",
+            data_asset_name="my_asset",
+            batch_identifiers=IDDict(),
+        )
```
Comment (joshua-stauffer) [inline great_expectations/self_check/util.py:922]: Previously, this function used the `ConfiguredAssetSqlDataConnector` to build a `LegacyBatchDefinition`. After running through a bunch of code, it appears that the batch definition never has `batch_identifiers`, and all the other fields are dummy fields not corresponding to real data. The solution is to just stand up the `LegacyBatchDefinition`, and tests appear to be satisfied.

File: tests/test_utils.py [hunk 3]
```diff
@@ -742,18 +742,7 @@ def clean_up_tables_with_prefix(connection_string: str, table_prefix: str) -> Li
...
-            "name": "temp_data_connector",
-        },
-        runtime_environment={
-            "execution_engine": execution_engine,
-            "datasource_name": "temp_datasource",
-        },
-        config_defaults={"module_name": "great_expectations.datasource.data_connector"},
-    )
-    introspection_output = data_connector._introspect_db()
+    introspection_output = introspect_db(execution_engine=execution_engine)
```
Comment (joshua-stauffer) [inline tests/test_utils.py:745]: This code previously used an `InferredAssetSqlDataConnector` to introspect the database and obtain table names for the test teardown process. `data_connector._introspect_db()` had no dependencies on the data connector class outside of the `SqlAlchemyExecutionEngine`, so I refactored that method to a stand alone function which requires the execution engine as an argument.

**Comment by joshua-stauffer** [comment]
while doing this work, I discovered that`test_sql_asset_workflows.py` depends on test data stood up by `test_expectations_v3_api.py`. It's unrelated to this PR, but will need to be cleaned up as part of https://greatexpectations.atlassian.net/browse/V1-347

**Pull request merged**